integration-tests/rest-client/src/main/java/io/quarkus/it/rest/client/selfsigned/ExternalSelfSignedResource.java

```diff
@@ -1,45 +1,29 @@
 package io.quarkus.it.rest.client.selfsigned;
 
 import java.io.IOException;
-import java.net.URL;
 
-import javax.net.ssl.HttpsURLConnection;
+import javax.inject.Inject;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
 /**
  * This has nothing to do with rest-client, but we add it here in order to avoid creating
  * a new integration test that would slow down our CI
  */
 @Path("/self-signed")
 public class ExternalSelfSignedResource {
 
+    @Inject
+    @RestClient
+    ExternalSelfSignedClient client;
+
     @GET
     @Produces(MediaType.TEXT_PLAIN)
     public String perform() throws IOException {
-        try {
-            return doGetCipher();
-        } catch (IOException e) {
-
-            // if it fails it might be because the remote service is down, so sleep and try again
-            try {
-                Thread.sleep(1000);
-            } catch (InterruptedException ignored) {
-
-            }
-
-            return doGetCipher();
-        }
-    }
-
-    private String doGetCipher() throws IOException {
-        // this URL provides an always on example of an HTTPS URL utilizing self-signed certificate
-        URL url = new URL("https://self-signed.badssl.com/");
-        HttpsURLConnection con = (HttpsURLConnection) url.openConnection();
-        con.setRequestMethod("GET");
-        con.getResponseCode();
-        return con.getCipherSuite();
```

Comment:
Actually, now that I think of it, I think that this block was meant to test that that the truststore is used by default. With the change it won't be doing that as the truststore is specific to the rest client

Reply:
Sounds like a test smell to me :)

Reply:
Maybe... But what other way do you have of testing that the truststore works in native mode?

Reply:
Well, setting the truststore specifically in the TestResource class for the rest client works in native, maybe that's how we should do it to keep tests isolated

Reply:
The thing is that for native, truststore isn't settable at runtime IIRC

Reply:
I am still against this change. The rest looks real good

Reply:
I am probably missing something, but how do you set the default trust store in this test? I don't see that being done in the existing test, can you point me to where that is set?

Reply:
I'll take a closer look tomorrow

Reply:
What I said above is indeed why it was done this way. And it works because of: https://github.com/quarkusio/quarkus/blob/22c529ec7b9739a24c152e5075caa4e19bd0ded5/integration-tests/rest-client/pom.xml#L105

So I really think this part of the PR should be reverted, otherwise we have no way of testing the default trust-store handling in native image

Reply:
@geoand this wasn't changed, the system properties are still set when the test runs and the specified trust-store is still loaded as the default in both tests. 

Note that there isn't a `QuarkusTestResourceLifecycleManager` associated with the `ExternalSelfSignedTestCase`, so the `self-signed.trust-store` is used only in the `javax.net.ssl.trustStore` property

Reply:
Yeah, but unless I am misreading the PR, now there is nowhere where the default trust-store is exercised. This is what the existing code is meant to do

Reply:
I see, but is this the right place to test this feature? I mean, this is not related to the rest-client integration, right?

Reply:
Yes and no. 

I added it here because this came about when I fixed the SSL handling a little while a back (it broke when we moved to 19.3).
And unless we have this sort of test, it can easily break again.

Sure there are perhaps better places to put it, but I don't really see a lot of value in moving it

Reply:
Cool, I'm convinced now. Thanks for the explanation :wink:

Reply:
Sure they 👌